src/core/lispStream.cc

```diff
@@ -1367,331 +992,110 @@ static int utf_8_encoder(T_sp stream, unsigned char* buffer, claspCharacter c) {
- */
-
-static cl_index clos_stream_read_byte8(T_sp strm, unsigned char* c, cl_index n) {
-  cl_index i;
-  for (i = 0; i < n; i++) {
-    T_sp byte = eval::funcall(gray::_sym_stream_read_byte, strm);
-    if (!core__fixnump(byte))
-      break;
-    c[i] = (byte).unsafe_fixnum();
-  }
-  return i;
-}
-
-static cl_index clos_stream_write_byte8(T_sp strm, unsigned char* c, cl_index n) {
-  cl_index i;
-  for (i = 0; i < n; i++) {
-    T_sp byte = eval::funcall(gray::_sym_stream_write_byte, strm, make_fixnum(c[i]));
-    if (!core__fixnump(byte))
-      break;
+cl_index FileStream_O::compute_char_size(claspCharacter c) {
+  // TODO  Make this work with full characters
+  unsigned char buffer[5];
+  int l = 0;
+  if (c == CLASP_CHAR_CODE_NEWLINE) {
+    if (_Flags & CLASP_STREAM_CR) {
+      l += encode(buffer, CLASP_CHAR_CODE_RETURN);
+      if (_Flags & CLASP_STREAM_LF)
+        l += encode(buffer, CLASP_CHAR_CODE_LINEFEED);
+    } else {
+      l += encode(buffer, CLASP_CHAR_CODE_LINEFEED);
+    }
+  } else {
+    l += encode(buffer, c);
   }
-  return i;
-}
-
-static T_sp clos_stream_read_byte(T_sp strm) {
-  T_sp b = eval::funcall(gray::_sym_stream_read_byte, strm);
-  if (b == kw::_sym_eof)
-    b = nil<T_O>();
-  return b;
-}
-
-static void clos_stream_write_byte(T_sp c, T_sp strm) { eval::funcall(gray::_sym_stream_write_byte, strm, c); }
-
-static claspCharacter clos_stream_read_char(T_sp strm) {
-  T_sp output = eval::funcall(gray::_sym_stream_read_char, strm);
-  gctools::Fixnum value;
-  if (cl__characterp(output))
-    value = output.unsafe_character();
-  else if (core__fixnump(output))
-    value = (output).unsafe_fixnum();
-  else if (output == nil<T_O>() || output == kw::_sym_eof)
-    return EOF;
-  else
-    value = -1;
-  unlikely_if(value < 0 || value > CHAR_CODE_LIMIT) FEerror("Unknown character ~A", 1, output.raw_());
-  return value;
-}
-
-static claspCharacter clos_stream_write_char(T_sp strm, claspCharacter c) {
-  eval::funcall(gray::_sym_stream_write_char, strm, clasp_make_character(c));
-  return c;
-}
-
-static void clos_stream_unread_char(T_sp strm, claspCharacter c) {
-  eval::funcall(gray::_sym_stream_unread_char, strm, clasp_make_character(c));
-}
-
-static claspCharacter clos_stream_peek_char(T_sp strm) {
-  T_sp out = eval::funcall(gray::_sym_stream_peek_char, strm);
-  if (out == kw::_sym_eof)
-    return EOF;
-  return clasp_as_claspCharacter(gc::As<Character_sp>(out));
+  return l;
 }
 
-static cl_index clos_stream_read_vector(T_sp strm, T_sp data, cl_index start, cl_index end) {
-  T_sp fn = eval::funcall(gray::_sym_stream_read_sequence, strm, data, clasp_make_fixnum(start), clasp_make_fixnum(end));
-  if (fn.fixnump()) {
-    return fn.unsafe_fixnum();
+T_sp FileStream_O::string_length(T_sp string) {
+  int l = 0;
+  if (cl__characterp(string)) {
+    l = compute_char_size(string.unsafe_character());
+  } else if (cl__stringp(string)) {
+    Fixnum iEnd;
+    String_sp sb = string.asOrNull<String_O>();
```

Comment:
Obviously this is just code you're copying and so fixing it is a bit out of scope, but just to mention (you probably already know):
1) asOrNull is to be avoided. we have `gc::As` etc
2) this conditional is pointless, since `cl__length` will return the fill pointer
3) ditto StringFillp and SetStringFillp. they should be on the chopping block